### ISSUE TITLE
Open the accessibility of all theme related classes and structures.

### ISSUE THREAD

**Issue by SF-Simon** [dotnet/maui#14455]
### Description

IStyleElement、MergedStyle、IStyle、IStyleSelectable

These interfaces are all protective, and we cannot use them when writing our own controls.

### Public API Changes

IStyleElement、MergedStyle、IStyle、IStyleSelectable

### Intended Use-Case

If they do not have bugs and other issues, please set to Public. Let's also try these methods when creating third-party controls, so that the controls are not hierarchical and feel deeply malicious.

**Comment by Eilon** [comment]
@SF-Simon could you also share examples of how you'd like to use these types if they were made public? That could help ensure that any future design changes could account for what you need.
